### ISSUE TITLE
Why CloseableAnimatedImage isn't be cache into BitmapMemoryCache?

### ISSUE THREAD

**Issue by huangjunkun** [facebook/fresco#1333]
Why CloseableAnimatedImage isn't be cache into BitmapMemoryCache?

> com.facebook.imagepipeline.producers.BitmapMemoryCacheProducer#wrapConsumer
> 
> ```
> ```

```
    // stateful results cannot be cached and are just forwarded
    if (newResult.get().isStateful()) {
      getConsumer().onNewResult(newResult, isLast);
      return;
    }
```

```

```

**Comment by oprisnik** [comment]
I suppose this is mainly because most animations would be quite big (and hence would clear everything else from the cache).

**Comment by oprisnik** [comment]
We're also working on a new version that will cache bitmaps in the default BitmapMemoryCache.

**Comment by huangjunkun** [comment]
@oprisnik  
Yeah, I've  extended imagepipeline for caching gif's Bitmaps，and It have been running normally.

**Comment by oprisnik** [comment]
We've released [Fresco v1.4.0](https://github.com/facebook/fresco/releases/tag/v1.4.0), which includes a new implementation for animated GIFs that is way more performant, and caches animation frames in BitmapMemoryCache.
